aten/src/ATen/ParallelNative.cpp

```diff
@@ -101,30 +150,46 @@ int get_num_threads() {
     TORCH_INTERNAL_ASSERT(nthreads == CONSUMED);
     return internal::_get_intraop_pool().size() + 1;
   }
+#else
+  caffe2::ThreadPool* pool = caffe2::mobile_threadpool();
+  // caffe2::ThreadPool::getNumThreads() counts the current thread.
+  return !pool ? 1 /* current thread */ : pool->getNumThreads();
+#endif // C10_MOBILE
 }
 
 int get_thread_num() {
   return thread_num_;
 }
 
 bool in_parallel_region() {
+#ifndef C10_MOBILE
   return in_parallel_region_ || (
     num_intraop_threads.load() == CONSUMED &&
     internal::_get_intraop_pool().inThreadPool()
   );
+#else
+  return in_parallel_region_;
+#endif // C10_MOBILE
 }
 
 void intraop_launch(std::function<void()> func) {
+#ifndef C10_MOBILE
   if (!in_parallel_region() && get_num_threads() > 1) {
     internal::_get_intraop_pool().run(func);
   } else {
     // execute inline if we're in parallel region
     func();
   }
+#else
+  // TODO: caffe2::ThreadPool doesn't support submitting tasks separately and
+  // running in parallel. Should fix it when this API becomes popular.
+  func();
```

Comment:
maybe also throw? or at least add some warning logging

Reply:
also same for intraop_launch_future

Reply:
The reason I don't throw for these APIs is that they are actually called from some unit tests - and we don't necessarily want to block other engs from using these APIs simplify because mobile doesn't support it yet. We could work on it separately as fixing "perf issues" instead of crashes...